nvidia_gpu_stats/src/metrics.rs

```diff
@@ -0,0 +1,36 @@
+use serde::Serialize;
+use std::collections::BTreeMap;
+
+/// A struct to store system metrics.
+/// Metrics are stored in a BTreeMap to ensure
+/// consistent ordering of keys in the output JSON.
+/// The output map is flat to make it easier to parse
+/// in downstream applications.
```

Comment:
formatting could be better here

https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text

* you don't need to say it's a struct
* separate the summary sentence from the details by a blank line
* the details can go up to ~80 characters I'd say. Using something like semantic line breaks (https://sembr.org/) is reasonable too, but I'd still use fewer line breaks even with that.